src/LndClient/Data/LndEnv.hs

```diff
@@ -62,69 +65,119 @@ data LndEnv
         envLndLogErrors :: Bool
       }
 
+instance ToGrpc LndWalletPassword ByteString where
+  toGrpc x = Right $ encodeUtf8 (coerce x :: Text)
+
+instance Read LndTlsCert where
+  readsPrec _ x =
+    case createLndTlsCert $ C8.pack x of
+      Right r -> [(r, "")]
+      Left _ -> case reads x of
+        [(r, "")] -> [(r, "")]
+        _ -> []
+
+instance FromJSON LndTlsCert where
+  parseJSON x =
+    case x of
+      A.String s ->
+        case createLndTlsCert $ C8.pack $ T.unpack s of
+          Right cert -> return cert
+          Left e -> failure e
+      e -> failure e
+    where
+      failure err = fail $ "Json certificate parsing error: " <> " " <> show err
+
+instance Read LndPort where
+  readsPrec _ x = do
+    let mX :: Maybe Word32 = readMaybe x
+    let eX :: Either LndError Word32 = maybeToRight (LndEnvError "Port parsing error") mX
+    case createLndPort =<< eX of
+      Right r -> [(r, "")]
+      Left _ -> case reads x of
+        [(r, "")] -> [(r, "")]
+        _ -> []
+
+instance FromJSON LndPort where
+  parseJSON x =
+    case x of
+      A.Number s -> do
+        let ePort = maybeToRight (LndEnvError "Port should be Int") $ toBoundedInteger s
+        case ePort >>= createLndPort of
+          Right lndPort -> return lndPort
+          Left err -> failure err
+      err -> failure err
+    where
+      failure err = fail $ "Json port loading error: " <> " " <> show err
+
+instance FromJSON RawConfig where
+  parseJSON (Object v) =
+    RawConfig <$> v .: "lnd_wallet_password"
+      <*> v .: "lnd_tls_cert"
+      <*> v .: "lnd_hex_macaroon"
+      <*> v .: "lnd_host"
+      <*> v .: "lnd_port"
+  parseJSON _ = mzero
```

Comment:
Interesting, have never seen this kind of implementation, let's also write a test to see how `FromJSON` will fail if given JSON will be not an object, might be separate PR.